streaming/src/main/scala/org/apache/spark/streaming/Checkpoint.scala

```diff
@@ -184,8 +184,7 @@ class CheckpointWriter(
   val executor = Executors.newFixedThreadPool(1)
   val compressionCodec = CompressionCodec.createCodec(conf)
   private var stopped = false
-  private var _fs: FileSystem = _
-
+  @volatile private[this] var fs: FileSystem = null
```

Comment:
Does this even need to be a member? it looks like it's used entirely in one method.